File: src/java/src/main/java/org/elastos/hive/vault/scripting/Condition.java
package org.elastos.hive.vault.scripting;

import com.google.gson.annotations.SerializedName;

public abstract class Condition {
	@SerializedName("name")
	private String name;
	@SerializedName("type")
	private String type;
	@SerializedName("body")
	private Object body;

	protected Condition(String name, String type, Object body) {
		this.name = name;
		this.type = type;
		this.body = body;
	}

	protected void setBody(Object value) {
		body = value;
	}

	protected Object getBody() {
		return body;
	}
}

File: src/restclient/scripting/condition.ts

export class Condition {
	private name: string;
	private type: string;
	private body: any;

	constructor(name: string, type: string, body: any) {
		this.name = name;
		this.type = type;
		this.body = body;
	}

	setName(value: any) {
		this.name = value;
	}

    getName() : any {
		return this.name;
	}

	setType(value: any) {
		this.type = value;
	}

    getType() : any {
		return this.type;
	}

	setBody(value: any) {
		this.body = value;
	}

    getBody() : any {
		return this.body;
	}
}
